libs/core/langchain_core/tools.py

```diff
@@ -784,31 +800,39 @@ def args(self) -> dict:
     def _run(
         self,
         *args: Any,
+        config: RunnableConfig,
         run_manager: Optional[CallbackManagerForToolRun] = None,
         **kwargs: Any,
     ) -> Any:
         """Use the tool."""
         if self.func:
             if run_manager and signature(self.func).parameters.get("callbacks"):
                 kwargs["callbacks"] = run_manager.get_child()
+            if config_param := _get_runnable_config_param(self.func):
```

Comment:
should we just do this once instead of on every invocation?

Reply:
i guess teeeechnically self.func is mutable